srcpkgs/mongoose/patches/fix-tests.patch

```diff
@@ -0,0 +1,149 @@
+-  for (i = 0; i < 300 && buf[0] == 0; i++) mg_mgr_poll(&mgr, 10);
+-  ASSERT(buf[0] == 'X');
+-  mg_mqtt_sub(c, topic, 1);
+-  mg_mqtt_pub(c, topic, data, 1, false);
+-  for (i = 0; i < 500 && buf[1] == 0; i++) mg_mgr_poll(&mgr, 10);
+-  ASSERT(strcmp(buf, "Xx/f12/hi") == 0);
+-
+-  mg_mgr_free(&mgr);
+-  ASSERT(mgr.conns == NULL);
+-}
+-
+ static void eh1(struct mg_connection *c, int ev, void *ev_data, void *fn_data) {
+   struct mg_tls_opts *topts = (struct mg_tls_opts *) fn_data;
+   if (ev == MG_EV_ACCEPT && topts != NULL) mg_tls_init(c, topts);
+@@ -750,63 +689,6 @@ static void test_tls(void) {
+ #endif
+ }
+ 
+-static void f3(struct mg_connection *c, int ev, void *ev_data, void *fn_data) {
+-  int *ok = (int *) fn_data;
+-  // LOG(LL_INFO, ("%d", ev));
+-  if (ev == MG_EV_CONNECT) {
+-    // c->is_hexdumping = 1;
+-    mg_printf(c, "GET / HTTP/1.0\r\nHost: %s\r\n\r\n",
+-              c->peer.is_ip6 ? "ipv6.google.com" : "cesanta.com");
+-  } else if (ev == MG_EV_HTTP_MSG) {
+-    struct mg_http_message *hm = (struct mg_http_message *) ev_data;
+-    // LOG(LL_INFO, ("-->[%.*s]", (int) hm->message.len, hm->message.ptr));
+-    // ASSERT(mg_vcmp(&hm->method, "HTTP/1.1") == 0);
+-    // ASSERT(mg_vcmp(&hm->uri, "301") == 0);
+-    *ok = atoi(hm->uri.ptr);
+-  } else if (ev == MG_EV_CLOSE) {
+-    if (*ok == 0) *ok = 888;
+-  } else if (ev == MG_EV_ERROR) {
+-    if (*ok == 0) *ok = 777;
+-  }
+-}
+-
+-static void test_http_client(void) {
+-  struct mg_mgr mgr;
+-  struct mg_connection *c;
+-  int i, ok = 0;
+-  mg_mgr_init(&mgr);
+-  c = mg_http_connect(&mgr, "http://cesanta.com", f3, &ok);
+-  ASSERT(c != NULL);
+-  for (i = 0; i < 500 && ok <= 0; i++) mg_mgr_poll(&mgr, 10);
+-  ASSERT(ok == 301);
+-  c->is_closing = 1;
+-  mg_mgr_poll(&mgr, 0);
+-  ok = 0;
+-#if MG_ENABLE_MBEDTLS || MG_ENABLE_OPENSSL
+-  {
+-    struct mg_tls_opts opts = {.ca = "./test/data/ca.pem"};
+-    c = mg_http_connect(&mgr, "https://cesanta.com", f3, &ok);
+-    ASSERT(c != NULL);
+-    mg_tls_init(c, &opts);
+-    for (i = 0; i < 500 && ok <= 0; i++) mg_mgr_poll(&mgr, 10);
+-    ASSERT(ok == 200);
+-  }
+-#endif
+-
+-#if MG_ENABLE_IPV6
+-  ok = 0;
+-  // ipv6.google.com does not have IPv4 address, only IPv6, therefore
+-  // it is guaranteed to hit IPv6 resolution path.
+-  c = mg_http_connect(&mgr, "http://ipv6.google.com", f3, &ok);
+-  ASSERT(c != NULL);
+-  for (i = 0; i < 500 && ok <= 0; i++) mg_mgr_poll(&mgr, 10);
+-  ASSERT(ok == 200);
+-#endif
+-
+-  mg_mgr_free(&mgr);
+-  ASSERT(mgr.conns == NULL);
+-}
+-
+ static void f4(struct mg_connection *c, int ev, void *ev_data, void *fn_data) {
+   if (ev == MG_EV_HTTP_MSG) {
+     struct mg_http_message *hm = (struct mg_http_message *) ev_data;
+@@ -1680,11 +1562,9 @@ int main(void) {
+   test_ws();
+   test_ws_fragmentation();
+   test_http_server();
+-  test_http_client();
+   test_http_no_content_length();
+   test_http_pipeline();
+   test_http_range();
+-  test_mqtt();
```

Comment:
If you're commenting out the `test_mqtt` and `test_http_client` here, you shouldn't need to remove all of the functions above.

Reply:
It complains _function defined but not used_

Reply:
Is that a hard failure, or a warning? It shouldn't prevent compilation.

Reply:
hard failure, because of `-Werror`

Reply:
Add `make_check_args="WARN="` above `make_check_target=test` and you can drop most of the patch.

Reply:
I dunno... it just bothers me to silence all warnings, even those that we didn't cause. :sweat_smile: 

Case in hand - building on musl causes this error to pop up -
https://github.com/void-linux/void-packages/runs/5748101481?check_suite_focus=true#step%3A7%3A54=

This error isn't present on glibc, so my guess is upstream has never tested on musl.  
If we disable all warnings, we won't be able to see these messages, and those message may show some errors that could develop to major problems.

Reply:
@ahesford I've added `CFLAGS+=" -Wno-error=unused-function"` to `pre_check()` in the latest commit, and pruned the `fix-tests.patch`.